jbpm-wb-process-runtime/jbpm-wb-process-runtime-api/src/main/java/org/jbpm/workbench/pr/events/ProcessInstanceSelectionEvent.java

```diff
@@ -44,6 +45,19 @@ public ProcessInstanceSelectionEvent(String serverTemplateId,
              forLog);
     }
 
+    public ProcessInstanceSelectionEvent(ProcessInstanceKey processInstanceKey,
+                                         boolean forLog,
+                                         boolean isDiagram) {
+
+        this.processInstanceKey = processInstanceKey;
+        this.forLog = forLog;
+        this.isDiagram = isDiagram;
+    }
+
+    public boolean isDiagram() {
```

Comment:
What does the method really do? Isn't there a better naming?
From what I see below, the usage is:
```
        if (event.isDiagram()) {
            view.showDiagramTab();
        }
```
so I would rather name the field differently .... but I can't think of any great solution. But my best guess - field toBeOpened and method isToBeOpened() 

@cristianonicolai @nmirasch do you have a better candidate? :-)

Reply:
@jstastny-cz @bxf12315  about the naming, something like diagramRefresh or diagramReload would work?
AFAIK, I think the point is not to know if you are able to navigate tho the diagram it's a matter or reloading the diagram when you are navigating between related pi

Reply:
IMO that's not a matter of reloading the diagram. The event itself triggers reload of the process instance details page as a whole. The field here is being used to navigate to Diagram tab afterwards.
I would look for a more general solution, e.g. enable the event to carry information about a tab that should be displayed initially (not a boolean field but some Tab identifier).

Reply:
My original version，the filed have five states, let it can delegate different tabs, but these tabs just process instances and diagram will throw the event. other tabs never use it, so I think general solution is unnecessary.

Reply:
@cristianonicolai @jstastny-cz @nmirasch For general solution, how about this?, add a new event ProcessInstanceDetailsTabSelectedEvent , it has five states. then remove isDiagram from ProcessInstanceSelectedEvent.

Reply:
IMO ProcessInstanceSelectionEvent is the correct place to make the change. We just might need to come up with a mechanism how to be able to specify target Tab via the event. Would an Enum with pre-defined values (e.g. DETAILS, LOGS, DIAGRAM,...) be a good solution? Plus a method in the event getTargetTab() that would return value from a field which would be set by default to DETAILS or overriden when value provided in constructor. Then the method in presenter onProcessSelectionEvent would show a correct Tab based on the enum value. WDYT? Is this useful for future enhancements or is the diagram tab only solution enough?

Reply:
I think it's maybe not a good idea, ProcessInstanceSelectionEvent from process instance list page, I think it's not related to any tabs of Process instance details page. Now, we set some Enum of tabs of Process instance details, it's not a good idea.

Reply:
Ok, let's keep the changes minimal ... I am fine with boolean field now, just the name needs better describe the purpose.

Reply:
ok. minimal change is good. How about "isFromDiagram"?

Reply:
Yep, that does work well - providing information from where the event has been fired. Fine with me.

Reply:
Done